pkg/solana/chainwriter/helpers.go

```diff
@@ -84,6 +84,22 @@ func errorWithDebugID(err error, debugID string) error {
 // traversePath recursively traverses the given structure based on the provided path.
 func traversePath(data any, path []string) ([]any, error) {
 	if len(path) == 0 {
+		val := reflect.ValueOf(data)
+
+		// If the final data is a slice or array, flatten it into multiple items,
+		if val.Kind() == reflect.Slice || val.Kind() == reflect.Array {
+			// don't flatten []byte
```

Comment:
Is it possible for us to infer if a byte array is actually a Solana PublicKey?
I'm thinking, theoretically, a byte can be a seed right?
So a []byte may also technically be a list of seeds sometimes. And other times, it can be a Solana Address

Reply:
Hm that's a good point, a single byte could potentially be a seed, and if we want each element in an array of bytes to be a seed, that wouldn't be possible with this current logic. I'm unsure what validation would make it possible to enable this kind of functionality and whether its worth adding complexity for a case we may not even need.

Reply:
Sounds good, yes it is ok to let things be as it is for now.